### PR TITLE
Refactor admin create-group views

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR refactors `h.views.admin.groups` create handlers, in order to:

* Use existing services where extant (e.g. user fetching) instead of direct model queries
* rename the class to reflect the reality that we don't have Controllers in pyramid
* increase consistency of how services are handled
* streamline where possible

I've refactored and minimized the tests as much as I can without deeper effort at this point. The architecture of our views + forms makes test configuration somewhat complex (the on-success callbacks end up being the majority of the view logic). I've added more mocks and tried to decouple things in tests. It's not perfect, but I hope it's better!

### CODE REVIEW

File: h/views/admin/groups.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def groups_index(context, request):
     renderer="h:templates/admin/groups_create.html.jinja2",
     permission="admin_groups",
 )
-class GroupCreateController(object):
+class GroupCreateView(object):
```
Comment (seanh) [inline h/views/admin/groups.py]: The class should be called `GroupCreateViews` (plural) because it's a class that groups multiple views (the methods) into one class. See <https://docs.pylonsproject.org/projects/pyramid/en/latest/quick_tutorial/view_classes.html>.

File: tests/h/views/admin/groups_test.py [hunk 2]
```diff
@@ -82,9 +82,9 @@ def test_index_filters_results(pyramid_request, factories, query, expected_group
 
 
 @pytest.mark.usefixtures("group_create_svc", "list_orgs_svc", "routes", "user_svc")
-class TestGroupCreateController(object):
+class TestGroupCreateView(object):
     def test_get_sets_form(self, pyramid_request):
-        ctrl = GroupCreateController(pyramid_request)
+        ctrl = GroupCreateView(pyramid_request)
```
Comment (seanh) [inline tests/h/views/admin/groups_test.py]: We should rename `ctrl` (controller) to `views` throughout this test class.

File: h/views/admin/groups.py [hunk 3]
```diff
@@ -46,24 +46,32 @@ def groups_index(context, request):
...
+    """View handlers for creating new groups"""
```
Comment (seanh) [inline h/views/admin/groups.py]:
```suggestion
     """Views for the admin create group form."""
```

They're just called "views" (or sometimes "view callables" or I guess view methods or view functions) in Pyramid. "View handler" means [something else](https://docs.pylonsproject.org/projects/pyramid_handlers/en/latest/).

File: h/views/admin/groups.py [hunk 4]
```diff
@@ -47,23 +47,31 @@ def groups_index(context, request):
...
     def __init__(self, request):
-        user_svc = request.find_service(name="user")
-        list_org_svc = request.find_service(name="list_organizations")
-        self.organizations = {o.pubid: o for o in list_org_svc.organizations()}
+        self.request = request
+
+        self.user_svc = self.request.find_service(name="user")
+        self.list_org_svc = self.request.find_service(name="list_organizations")
+        self.group_create_svc = self.request.find_service(name="group_create")
+        self.group_members_svc = self.request.find_service(name="group_members")
```
Comment (seanh) [inline h/views/admin/groups.py]: Yep, I think this is a nicer way of doing this than having each method retrieve them itself (and takes advantage of the fact that these methods are grouped into a class).

File: h/views/admin/groups.py [hunk 5]
```diff
@@ -46,24 +46,32 @@ def groups_index(context, request):
...
-            request=request, organizations=self.organizations, user_svc=user_svc
+            request=request, organizations=self.organizations, user_svc=self.user_svc
         )
-        self.request = request
         self.form = _create_form(self.request, self.schema, (_("Create New Group"),))
 
     @view_config(request_method="GET")
     def get(self):
-
+        """Render create-group form"""
```
Comment (seanh) [inline h/views/admin/groups.py]:
```suggestion
         """Render the admin create group form."""
```

Nice to have a docstring though!

File: h/views/admin/groups.py [hunk 6]
```diff
@@ -101,24 +108,16 @@ def on_success(appstruct):
...
-            member_userids = [
-                _userid(username, organization.authority)
-                for username in appstruct["members"]
-            ]
-            group_members_svc.add_members(group, member_userids)
+            member_userids = []
+            for username in appstruct["members"]:
+                member_userids.append(
+                    self.user_svc.fetch(username, organization.authority).userid
+                )
```
Comment (seanh) [inline h/views/admin/groups.py]: I think this could be:

```python
member_userids = [self.user_svc.fetch(username, organization.authority).userid
                  for username in appstruct["members"]
```

Which might be a little clearer and more succinct.

Comment (lyzadanger) [reply h/views/admin/groups.py]: I love a list comprehension, but this one was getting hard to parse IMHO.
